examples/c/src/L2_c_example.c

```diff
@@ -0,0 +1,105 @@
+#include <cuvs/core/c_api.h>
+#include <cuvs/distance/pairwise_distance.h>
+#include <cuvs/neighbors/cagra.h>
+
+#include <dlpack/dlpack.h>
+
+#include <cuda_runtime.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#define DIM 4
+#define N_ROWS 1
+
+float PointA[N_ROWS][DIM] = {1.0,2.0,3.0,4.0};
+float PointB[N_ROWS][DIM] = {2.0,3.0,4.0,5.0};
+
+void outputVector(float * Vec) {
+  printf("Vector is ");
+  for (int i = 0; i < DIM; ++i){
+    printf(" %f",Vec[i]);
+  }
+  printf("\n");
+}
+
+cuvsResources_t res;
+
+int initResources(void){
+  // Create a cuvsResources_t object
+  cuvsResourcesCreate(&res);
```

Comment:
Since these are one-liners anyways, it reduces the size of the code and clarity of the example to keep these inline in the example. Another small nitpick: we have intentionally named the C API functions consistently- `cuvsNounVerb` and changing that to `verbNoun` breaks this consistently. I'd keep `initializeTensor` below but rename it to `tensor_initialize` just to separate the casing from the C API. I'd also rename `calcL2Distance` to `l2_distance_calc`